conda/base/context.py

```diff
@@ -1201,6 +1205,13 @@ def description_map(self):
                 The list of conda channels to include for relevant operations.
                 """
             ),
+            channel_settings=dals(
+                """
+                A list of mappings that allows overriding certain settings for a single channel.
+                Each list item should include at least the "channel" key and the setting you would
```

Comment:
Since this PR can be seen as a first "reserve the config key" step, I'm okay if the following is done in a subsequent PR (but before `channel_settings` sees usage):
Some checks should be done while reading the configuration:
  - If `channel` key is not present, either
    1. error or
    2. warn and discard

    the entry (IDK/haven't thought about what's preferred here).
- In case of duplicate entries with the same `channel` key in the same file,
  1. error/warn, and
  2. if only warn, decide if first or last entry should be chosen.
- In case of duplicate entries with the same `channel` key in *different* files, decide if we want to
  1. override settings in higher priority config files, or
  2. merge the channel's setting keys.

  (the first option is easier to reason about).

Reply:
Thanks for the feedback @mbargull.

I will open an issue with this information in it to make sure it is addressed before we begin using this feature.